core/state_processor.go

```diff
@@ -401,10 +401,13 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 	posa, isPoSA := p.engine.(consensus.PoSA)
 	commonTxs := make([]*types.Transaction, 0, txNum)
 
-	// initilise bloom processors
+	// initialise bloom processors
 	bloomProcessors := NewAsyncReceiptBloomGenerator(txNum)
 	statedb.MarkFullProcessed()
 
+	// do trie prefetch for the big state trie tree in advance based transaction's From/To address.
+	statedb.TriePrefetchInAdvance(block, signer)
```

Comment:
Can you try to make it a asynchronous go routine?

Reply:
triePrefetcher's API is not thread safe, should be called in a same routine.

Reply:
That is a problem then, any proof that it is worthy to introduce it in the critical path?

Reply:
Good point, we will consider to redesigned the current triePrefetch to make `prefetch` concurrent safe.
I will hold this PR until the new tirePrefetch is ready

Reply:
updated, pls check